Feature/cleanup

geek3000 opened the pull request.

geek3000 committed:
Add reset and cleanup endpoints for namespace management

- Implemented `/reset` endpoint to delete and recreate namespaces for all provisioned users.
- Implemented `/cleanup` endpoint to remove resources for old provisioned users.
- Added utility functions to retrieve provisioned and old provisioned users from Keycloak.
- Updated user creation to include management attributes in the user profile.
- Enhanced Kubernetes namespace template with management labels.

geek3000 committed:
Add debug prints for user retrieval in provisioner

- Added print statements to log the list of provisioned users and old provisioned users for debugging purposes.
- Removed unnecessary Keycloak configuration code from the `get_old_provisioned_users` function to streamline the user retrieval process.

geek3000 committed:
Refactor provisioner and enhance Kubernetes cronjob templates

- Removed debug print statements for user retrieval in the provisioner to clean up the code.
- Added annotations and service account configuration to Kubernetes cronjob templates for improved security and secret management.
- Updated authorization header syntax in curl commands to ensure proper token usage.

geek3000 committed:
Update README with new API endpoints and automated tasks

- Modified the deletion command to include full name and email in the request body.
- Added new sections for resetting namespaces and cleaning up old users, detailing their respective API calls.
- Introduced automated tasks for monthly namespace resets and yearly user cleanups, outlining their schedules and purposes.
- Included testing instructions for various API endpoints to facilitate user testing.

imphilippesimo commented on README.md:
```diff
@@ -22,16 +22,54 @@ This will create :
+- Ensures clean state for all users
 
+### Yearly User Cleanup
+- Runs at midnight on December 31st
```
Must run eveyday to delete workspaces as soon as it is more that one year old.

If you wait on December 31, an account that turned one on march31st  would have had march<->December (9 months) for free.

Nevertheless, if it runs everyday, the account will be deleted on April 1st.

imphilippesimo commented on app/utils.py:
```diff
@@ -44,7 +44,11 @@ def create_keycloak_user(username, email):
         'username': username,
-        'credentials': [{'type': 'password', 'value': generated_password}]
+        'credentials': [{'type': 'password', 'value': generated_password}],
+        'attributes': {
```
Comment allons-nous tagguer tous les users déjà existant sur keycloak ? On en discute de vive voix ce soir.

imphilippesimo commented on app/k8s_templates/cronjobs.yaml:
```diff
@@ -0,0 +1,69 @@
+  name: cleanup-old-users
+  namespace: zerofiltretech-${env_name}
+spec:
+  schedule: "0 0 31 12 *"  # Run at midnight on December 31st
```
Must run everyday that GOD has created.

geek3000 committed:
Update user cleanup schedule in README and cronjob template

- Changed the cleanup task from a yearly schedule to a daily schedule in both the README and the Kubernetes cronjob template.
- Updated documentation to reflect the new frequency of user cleanup operations.
